Integrate gutenberg-mobile release 1.58.0

SiobhyB opened the pull request:
## Description
This PR incorporates the 1.58.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3746

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

SiobhyB committed:
Release script: Update build.gradle gutenbergMobileVersion to ref

SiobhyB committed:
Update reference to Gutenberg Mobile

SiobhyB committed:
Update RELEASE-NOTES.txt

SiobhyB committed:
Merge branch 'develop' into gutenberg/integrate_release_1.58.0

SiobhyB committed:
Update reference to Mobile Gutenberg

SiobhyB committed:
Merge branch 'gutenberg/integrate_release_1.58.0' of github.com:wordpress-mobile/WordPress-Android into gutenberg/integrate_release_1.58.0

fluiddot commented on RELEASE-NOTES.txt:
```diff
@@ -2,6 +2,7 @@
 
 17.9
 -----
+* [***] New Block: Embed block. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3727]
```
Not sure if it's standard but I noticed that other entries related to changes in the editor include `Block editor: ` message at the beginning, what do you think about including it too?

SiobhyB replied:
That's a good point, I've added that in https://github.com/wordpress-mobile/WordPress-Android/pull/15054/commits/a4c0328130180d967eebb394eb4c03fac2e1b575, to bring this set of release notes in line with previous releases. 🙇‍♀️

SiobhyB committed:
Add "Block editor" indicator to RELEASE-NOTES

This brings the notes in line with previous releases, which have included a "Block editor" indicator with any changes coming from Mobile Gutenberg.

SiobhyB committed:
Merge branch 'develop' into gutenberg/integrate_release_1.58.0

SiobhyB committed:
Update Mobile Gutenberg reference

SiobhyB committed:
Merge branch 'gutenberg/integrate_release_1.58.0' of github.com:wordpress-mobile/WordPress-Android into gutenberg/integrate_release_1.58.0

fluiddot commented:
⚠️ After I open and close a post/page, the next time I open a post/page the editor gets stuck loading. I debugged the build and looks like it's failing with the following error:
```
TypeError: Cannot read property 'replace' of undefined
    
    This error is located at:
        in R
        in WithDispatch(R)
        in Unknown
        in WithSelect(WithDispatch(R))
        in k
        in S
        in WithDispatch(S)
        in Unknown
        in WithSelect(WithDispatch(S))
        in P
        in RCTView
        in Unknown
        in RCTView
        in Unknown
        in C, js engine: hermes
``` 

We should investigate this issue further before proceeding with the release.

SiobhyB commented:
Thank you for spotting that, Carlos. 🙇‍♀️ I'm being pointed to [this file](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/mobile/global-styles-context/utils.native.js) when I replicate locally:

![Screenshot_1626963083](https://user-images.githubusercontent.com/2998162/126653836-e679e39a-f9ee-4fc3-ab33-eab6f990f7ca.png)

@antonis, I was just wondering whether this may be related to the recent changes you made in https://github.com/wordpress-mobile/WordPress-Android/pull/15050?

antonis commented:
> @antonis, I was just wondering whether this may be related to the recent changes you made in #15050?

Thank you for the ping @SiobhyB. This indeed seems related since it is enables the Global Styles feature. Looking at it 👀

SiobhyB committed:
Merge branch 'develop' into gutenberg/integrate_release_1.58.0

fluiddot commented on RELEASE-NOTES.txt:
```diff
@@ -2,6 +2,8 @@
 17.9
 -----
+* [***] Block editor: New Block: Embed block. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/3727]
+* [*] Enables Support for Global Style Colors with Full Site Editing Themes [https://github.com/wordpress-mobile/WordPress-Android/pull/15002]
```
We should probably remove this entry as [this PR](https://github.com/wordpress-mobile/WordPress-Android/pull/15058) disabled it.

SiobhyB replied:
Done in https://github.com/wordpress-mobile/WordPress-Android/pull/15054/commits/a8acc247e98c0dc7ca656ce3a6174908980e3b84.

fluiddot replied:
Thanks 🙇 !

SiobhyB committed:
Update RELEASE-NOTES

Remove reference to feature that's since been reverted.

fluiddot reviewed (approved):
LGTM 🎊 !

I did a quick test (build `112883`) with Samsung Galaxy S20 FE 5G (Android 10) and worked fine. I also confirm that the freezing issue is no longer happening 🥳 .

**NOTE:** Before merging this PR, we have to change the Gutenberg Mobile ref to point to the release tag `v1.58.0`.

SiobhyB committed:
Update Gutenberg mobile reference
